shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -141,20 +148,29 @@ class JSBackend(pgrm: Pgrm) {
       JSMember(translateTerm(receiver), fieldName.name)
     // Turn let into an IIFE.
     case Let(isRec, name, value, body) =>
-      JSImmEvalFn(name :: Nil, Left(translateTerm(body)), translateTerm(value) :: Nil)
+      val letScope = Scope(name :: Nil, scope)
+      JSImmEvalFn(
+        name :: Nil,
+        Left(translateTerm(body)(letScope)),
+        translateTerm(value)(letScope) :: Nil
+      )
     case Blk(stmts) =>
+      val blockScope = Scope(Nil, scope)
```

Comment:
Is it useful to create a new scope here? What does it achieve?

Reply:
We will translate `Let` to real `let`-declaration instead of IIFE, which needs to allocate names in the scope of current block. So creating scopes for blocks is necessary.

Reply:
Will `let x = 1 in let x = x in x` be translated correctly?

Reply:
> Will `let x = 1 in let x = x in x` be translated correctly?

Yes. The scope will be able to rename identifiers, which will address this kind of problems.

Reply:
You keep saying "will", as this is not currently done, right? So for now this `blockScope` is useless and is better removed, or at least a comment stating its future purpose should be added.

Reply:
OK.